common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -489,27 +501,31 @@ def _render_student_view_for_items(self, context, display_items, fragment, view=
             show_bookmark_button = False
             is_bookmarked = False
 
-            if is_user_authenticated:
+            if is_user_authenticated and render_items:
                 show_bookmark_button = True
                 is_bookmarked = bookmarks_service.is_bookmarked(usage_key=usage_id)
 
             context['show_bookmark_button'] = show_bookmark_button
             context['bookmarked'] = is_bookmarked
 
-            rendered_item = item.render(view, context)
-            fragment.add_fragment_resources(rendered_item)
-
+            if render_items:
+                rendered_item = item.render(view, context)
+                fragment.add_fragment_resources(rendered_item)
+                content = rendered_item.content
+            else:
+                content = ''
             iteminfo = {
-                'content': rendered_item.content,
+                'content': content,
                 'page_title': getattr(item, 'tooltip_title', ''),
                 'type': item_type,
                 'id': text_type(usage_id),
                 'bookmarked': is_bookmarked,
                 'path': " > ".join(display_names + [item.display_name_with_default]),
                 'graded': item.graded
             }
-
-            if is_user_authenticated:
+            if not render_items:
+                iteminfo['href'] = '/container/%s' % text_type(usage_id)
```

Comment:
Can we do a lookup with reverse() here as well? Hardcoding the front of the URL makes it harder to use Sequence in the XBlock workbench or extract it as an XBlock.

Reply:
Using `reverse()` here would mean importing from django. I don't know if that's kosher for xmodules. I've changed it to use a context variable as the url template.

Reply:
We once had the notion that we'd keep XModule/XBlock free of Django dependencies, but we've long since given up on that.